### PR TITLE
Update UserCreateRequest password property xml documentation

### LINKED ISSUE #496: UserCreateRequest model password property XML documentation change
### Describe the problem
[The UserCreateRequest XML documentation](https://github.com/auth0/auth0.net/blob/master/src/Auth0.ManagementApi/Models/UserCreateRequest.cs#L23) is missing the email connection type from the summary as it is also non manditory. It currently says:

> Gets or sets the user's password. This is mandatory on non-SMS connections.

But the password property is also not manditory on email connections. My suggestion is to change it to:

> Gets or sets the user's password. This is mandatory on non SMS or email connections.

PR: #497 

Please let me know if you think it could be reworded better :)

### PR DESCRIPTION
**Pull request by Goolyio**
Closes #496 

Updated the password property XML documentation summary to include email as a non mandatory connection type.

### CODE REVIEW

**Comment by frederikprijck** [comment]
As mentioned in https://github.com/auth0/auth0.net/issues/496, holding off on reviewing until the API docs are updated.

File: src/Auth0.ManagementApi/Models/UserCreateRequest.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public class UserCreateRequest : UserBase
         public string UserId { get; set; }
 
         /// <summary>
-        /// Gets or sets the user's password. This is mandatory on non-SMS connections.
+        /// Gets or sets the user's password. This is mandatory on non SMS or email connections.
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Models/UserCreateRequest.cs:23]: Soon the docs will contain the following:

> Initial password for this user (mandatory only for auth0 connection strategy).

So lets update this to be:

```suggestion
/// Gets or sets the user's password. This is mandatory only for auth0 connection strategy.
```

**Review by frederikprijck — CHANGES_REQUESTED** [review]
Sorry for taking so long. I went and asked to get this resolved on the docs side and managed to get things progressing.

**Review by frederikprijck — APPROVED** [review]

**Comment by Goolyio** [comment]
> Sorry for taking so long. I went and asked to get this resolved on the docs side and managed to get things progressing.

NW I like your commit, makes a lot of sense. What are the next steps to getting this merged, is there anything you need from me?

**Comment by frederikprijck** [comment]
@Goolyio Nope, All is fine. Will merge it somewhere today.  Thanks again for the patience and the initial PR!

**Pull request merged**